src/main/java/io/confluent/connect/jdbc/source/DataConverter.java

```diff
@@ -289,19 +325,31 @@ private static void convertFieldValue(ResultSet resultSet, int col, int colType,
 
       // 8 bits int
       case Types.TINYINT: {
-        colValue = resultSet.getByte(col);
+        if (signed) {
+          colValue = resultSet.getByte(col);
+        } else {
+          colValue = resultSet.getShort(col);
+        }
         break;
       }
 
       // 16 bits int
       case Types.SMALLINT: {
-        colValue = resultSet.getShort(col);
+        if (signed) {
+          colValue = resultSet.getShort(col);
+        } else {
+          colValue = resultSet.getInt(col);
+        }
         break;
       }
 
       // 32 bits int
       case Types.INTEGER: {
-        colValue = resultSet.getInt(col);
+        if (resultSet.getMetaData().isSigned(col)) {
```

Comment:
Alternately it may make more sense to inline all the usages to avoid it being unnecessarily evaluated in some of the cases (e.g. if a JDBC driver throws an exception if you call `resultSet.getMetaData().isSigned(col)` for textual data)